File: Boyoung/추억점수.py
def solution(name, yearning, photo):
    total_yearn = [0] * len(photo)
    
    for i in range(len(photo)):
        for j in range(photo[i]):
            for k in range(len(name)):
                if name[k] == photo[i][j]:
                    total_yearn[i] += yearning[k]

    return list(total_yearn)

File: JoonSeong/추억 점수.java
import java.util.*;
import java.io.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        HashMap<String, Integer> score = new HashMap();
        for(int i = 0; i < name.length; i++) {
            score.put(name[i], yearning[i]);
        }
        int[] answer = new int[photo.length];
        
        for(int i = 0; i < photo.length; i++) {
            answer[i] = 0;
            for(int j = 0; j < photo[i].length; j++) {
                if(score.containsKey(photo[i][j])) {
                answer[i] += score.get(photo[i][j]);
                }
            }
        }
        return answer;
    }
}